.buildkite/pipeline.yml

```diff
@@ -13,160 +13,166 @@ steps:
-    parallelism: 3
-    artifact_paths:
-      - junit-*.xml
-      - "coverage/**/*"
-    agents:
-      queue: $AGENT_RUNNERS_LINUX_ARM64_QUEUE
-    plugins:
-      - docker-compose#v4.14.0:
-          config: .buildkite/docker-compose.yml
-          cli-version: 2
-          propagate-environment: true
-          run: agent
-      - test-collector#v1.11.0:
-          files: "junit-*.xml"
-          format: "junit"
-          tags:
-            - "os=linux"
-            - "arch=arm64"
-            - "race=true"
-
-  - name: ":windows: Windows AMD64 Tests"
-    key: test-windows
-    command: "bash .buildkite\\steps\\tests.sh"
-    parallelism: 2
-    artifact_paths:
-      - junit-*.xml
-      - "coverage/**/*"
-    agents:
-      queue: $AGENT_RUNNERS_WINDOWS_QUEUE
-    plugins:
-      - test-collector#v1.11.0:
-          files: "junit-*.xml"
-          format: "junit"
-          tags:
-            - "os=windows"
-            - "arch=arm64"
-            - "race=false"
-
-  - name: ":coverage: Test coverage report Linux ARM64"
-    key: test-coverage-linux-arm64
-    command: ".buildkite/steps/test-coverage-report.sh"
-    artifact_paths:
-      - "cover.html"
-      - "cover.out"
-    depends_on:
-      - test-linux-arm64
-    plugins:
-      - docker-compose#v4.14.0:
-          config: .buildkite/docker-compose.yml
-          cli-version: 2
-          run: agent
-      - artifacts#v1.9.4:
-          download: "coverage/**"
-          step: test-linux-arm64
-
-  - name: ":coverage: Test coverage report Linux AMD64"
-    key: test-coverage-linux-amd64
-    command: ".buildkite/steps/test-coverage-report.sh"
-    artifact_paths:
-      - "cover.html"
-      - "cover.out"
-    depends_on:
-      - test-linux-amd64
-    plugins:
-      - docker-compose#v4.14.0:
-          config: .buildkite/docker-compose.yml
-          cli-version: 2
-          run: agent
-      - artifacts#v1.9.4:
-          download: "coverage/**"
-          step: test-linux-amd64
-
-  - name: ":coverage: Test coverage report Linux ARM64 Race"
-    key: test-coverage-linux-arm64-race
-    command: ".buildkite/steps/test-coverage-report.sh"
-    artifact_paths:
-      - "cover.html"
-      - "cover.out"
-    depends_on:
-      - test-race-linux-arm64
-    plugins:
-      - docker-compose#v4.14.0:
-          config: .buildkite/docker-compose.yml
-          cli-version: 2
-          run: agent
-      - artifacts#v1.9.4:
-          download: "coverage/**"
-          step: test-race-linux-arm64
-
-  - label: ":writing_hand: Annotate with Test Failures"
-    depends_on:
-      - test-linux-amd64
-      - test-race-linux-arm64
-      - test-linux-arm64
-      - test-windows
-    allow_dependency_failure: true
-    plugins:
-      - junit-annotate#v1.6.0:
-          artifacts: junit-*.xml
+  - group: ":go::scientist: Tests and Coverage"
+    if_changed: "{go.mod,go.sum,**.go,**/fixtures/**,.buildkite/steps/{tests,test-coverage-report}.sh}"
```

Comment:
its great that you can use it at the group level 👨‍🍳 😚

Reply:
Oh yeah it was my first thought when implementing: "this should totally work on groups and trigger steps too"